yesod-core/src/Yesod/Core/Types.hs

```diff
@@ -306,6 +308,24 @@ data Content = ContentBuilder !BB.Builder !(Maybe Int) -- ^ The content and opti
              | ContentFile !FilePath !(Maybe FilePart)
              | ContentDontEvaluate !Content
 
+-- | Represents a Content as a String, rendering at most a specified number of
+-- bytes of the content, and annotating it with the remaining length.
+--
+-- @since 1.6.28.0
+contentToTruncatedString :: Content -> I.Int64 -> String
+contentToTruncatedString (ContentBuilder builder maybeLength) maxLength =
+    let
+      truncated = (T.unpack . Data.Text.Encoding.decodeUtf8) $ L.toStrict $ L.take maxLength (BB.toLazyByteString builder)
+      excess = case maybeLength of
+        (Just length) -> length - (fromIntegral maxLength)
+        Nothing -> 0
+    in case (excess > 0) of
```

Comment:
Just my curiosity, is it a code formatter that surrounds `Just length`, `excess > 0` and `fromIntegral maxLength` with parens or is it your personal preference?

Reply:
It's less a preference and more a habit inherited from working in other languages; I'm pretty new to Haskell.

I don't really think the parens aid legibility in the examples you pointed out, so I'll probably remove them.